testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -177,3 +189,22 @@ def test_presence_griddata(self):
             except ImportError:
                 self.fail('''MDAnalysis.analysis.density should not raise
                              an ImportError if gridData is available.''')
+
+
+#issue 1231 test for exception
+def test_check_set_unit():
+    import MDAnalysis.analysis.density
+    nbins = 3, 4, 5
+    counts = 100
+    Lmax = 10
+    bins = [np.linspace(0, Lmax, n+1) for n in nbins]
+    h, edges = np.histogramdd(
+    Lmax * np.sin(np.linspace(0, 1, counts * 3)).reshape(counts, 3),
+        bins=bins)
+    D = MDAnalysis.analysis.density.Density(h, edges,
```

Comment:
Here, we do not need a functional instance of `MDAnalysis.analysis.density.Density`; we only need an instance that gives us access to the `_check_set_unit` method. This means we could reduce the complexity of the test by a much simpler input. For instance, I managed to create an instance with the following code:

```python
import MDAnalysis.analysis.density as mad
mad.Density(numpy.zeros((2, 2)), numpy.zeros((2, 2)))
```